src/Shape.cpp

```diff
@@ -100,11 +100,57 @@ double line_parameter(const Vector3 &D, const Vector3 &v1, const Vector3 &v2, co
     return p_v1 + (p_v2-p_v1) * d_v1/(d_v1-d_v2);
 }
 
+double crossProduct(const Vec2d& v1, const Vec2d& v2) {
```

Comment:
Better to add this a member function to `Vec2T` in `math_types.h` I would call it `cross`.